rfcs/0123-flake-names.md

```diff
@@ -0,0 +1,91 @@
+---
+feature: flake-names
+start-date: 2022-03-12
+author: Anselm Schüler
+co-authors: None
+shepherd-team: None
+shepherd-leader: None
+related-issues: None
+---
+
+# Summary
+[summary]: #summary
+
+Flakes can declare the field `name`.  
+It represents the name of the flake.  
+The store paths corresponding to a flake source are no longer called `source`, but use the flake name.
+
+# Motivation
+[motivation]: #motivation
+
+- Flakes generate store paths named “source”, and it’s difficult to navigate this when manually inspecting the store.
+- This metadata can be used to make flakes more approachable and usable, in particular, it can be output instead or with the URI, making the nix commands friendlier.
+
```

Comment:
```suggestion
- Unique names are useful for the purpose of finding duplicate things caused by diamond dependencies. (E.g. `a -> b -> d1`, `a -> c -> d2`. Without a unique name (`d`), it is not possible to correlate `d1` and `d2` reliably.)

```

 - Most relevant when https://github.com/NixOS/nix/issues/5570 remains unimplemented. "Userspace" experimentation may yield some insights into solutions for 5570.